packages/client-python/src/rocketride/client.py

```
@@ -212,14 +229,44 @@ async def __aenter__(self):
+        instance before connecting.  If ROCKETRIDE_URI was set but is
+        unreachable, falls back to auto-spawn transparently.
+
         Returns:
             self: The connected client instance
         """
+        if self._engine_manager:
+            # If a URI is already set (from ROCKETRIDE_URI env), verify it's
+            # reachable before using it — fall back to auto-spawn if not.
+            env_uri = self._env.get('ROCKETRIDE_URI', '')
+            if self._uri and env_uri:
+                if not await self._engine_manager._is_engine_healthy_uri(env_uri):
```

Ruff (0.15.7)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)